Quiz clean up

treykris opened the pull request:
Description
-----------
- Encapsulated the different view pages from `QuizView.vue` into a `QuizResults.vue`, `QuizQuestions.vue`, and `QuizReview.vue` into their own components and changed `QuizView.vue` to `QuizView/index.vue`. 
- When a user submits their quiz a loader displaying `Getting your quiz results...` pops up.
- Updated styling and is fully responsive on mobile devices.




Note - a few improvements can still be made:
- Make a global CSS style for the styles copied across multiple quiz components
- Update the wrapper / change the structure to the wrapper that contains the new quiz view components


Developer self-review checklist
-------------------------------
- [x] Potentially confusing code has been explained with comments
- [x] No warnings or errors have been introduced; all known error cases have been handled
- [x] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [x] All edge cases have been addressed

jkeat commented on src/views/QuizView/index.vue:
```diff
@@ -0,0 +1,276 @@
+        <loading-message message="Getting your quiz results" />
+      </div>
+      <div v-else>
+        <div v-if="showNoQuiz" class="instructions">
```
Easter egg :o

jkeat commented on src/views/QuizView/index.vue:
```diff
@@ -0,0 +1,276 @@
+<template>
+  <div
+    v-if="user.isVolunteer && tries < 3"
```
Could you add a quick little `<div v-else>` with a message saying something like "no more tries" ?

treykris commented:
@jkeat,

Thank you! 

Just updated: added the message to tell a user they do not have any tries left

I just noticed that we might want to also add a message for users that try to visit a URL for a quiz that they are already certified in. This is a minor edge case
